neuraxle/base.py

```diff
@@ -228,28 +419,59 @@ def get_name(self) -> str:
         """
         return self.name
 
+    def get_savers(self) -> List[BaseSaver]:
+        """
+        Get the step savers of a pipeline step.
+
+        :return:
+        """
+        return self.savers
+
     def set_hyperparams(self, hyperparams: HyperparameterSamples) -> 'BaseStep':
+        self.is_invalidated = True
```

Comment:
added a warning :  warnings.warn('set_hyperparams was called on an already initialized(setup method called) step:  {0}:{1}. You probably wanted to set_hyperparams earlier instead of now ?'.format(self.name, self.__class__.__name__))